cfpb/cfgov-refresh
# Move credit card agreements to jinja

We have redundant references to chosen.js, which should ultimately be removed. As a step toward that, this moves the credit card agreement database page to jinja so we can use the `base.html` template and incorporate CF components.

There are number of opinionated edits in this PR that someone from UX would need to OK. These include:

- Replace custom modal with expandable.
- Create consistent search helper text ("Can't find an issuer?") between agreement landing page and search results page sidebar.
- Create consistent database disclosure and social media share icons at the bottom of both the agreement landing page and search results page sidebar.

## Changes

- Moves agreements app templates to jinja.
- Replace custom modal with expandable and add `single.js` to initialize expandables on /credit-cards/ URL.
- Replace custom PDF download arrow with svg download icon.
- Replaces search results table with cf-table.
- Replace pagination with cf-pagination.
- Export BASE_CLASS from Multiselect for initialization purposes (not yet used, because chosen is still used).

## Testing

1. `./frontend.sh`
2. Visit http://localhost:8000/credit-cards/agreements/ and ensure "Can't find issuer" expandable expands. 
3. Make sure links work on the page.
4. Select chosen.js drop-down of banks to go to the search results page.
5. Perform same checks on search results page.
6. Search for "Citibank" to get a page with pagination at the bottom of the table.
7. Click through pagination pages.

## Screenshots

![download-1](https://user-images.githubusercontent.com/704760/52822859-f112d080-3080-11e9-93a9-fb196ac59022.png)

![download-2](https://user-images.githubusercontent.com/704760/52822860-f112d080-3080-11e9-85af-7f476d6d4700.png)

![download-3](https://user-images.githubusercontent.com/704760/52822861-f112d080-3080-11e9-9b14-53dfe608bd46.png)



## Todos

- Chosen.js needs to be replaced with something, though the multiselect does not currently support anything other than 5 options selected.
- Tests are probably broken.
